apis/v1alpha2/gateway_types.go

```diff
@@ -525,6 +525,12 @@ const (
 	//
 	// Support: Implementation-specific
 	NamedAddressType AddressType = "NamedAddress"
+
+	// A hostname represents a DNS based ingress points. For example, this
+	// concept may be used for AWS load-balancers.
+	//
+	// Support: Implementation-specific
+	HostnameAddressType AddressType = "Hostname"
```

Comment:
Is it worth calling out explicitly the different in hostname vs NamedAddress? I think I understand it now but only when looking at the two fields docs side by side

Reply:
Can we switch the ordering here a bit?
I expect IPAdressType and HostnameAddressType to be the more common one. In that order, NamesAddress creates less confusion too.

Reply:
Good call, I've tried to update the `NamedAddress` type to be a bit less confusing and moved it after hostname to more clearly match how frequently the types will be used. Hopefully that helps. Open to any other ways to clarify the difference between these types.